pkg/api/v1/resource/helpers.go

```diff
@@ -24,36 +24,25 @@ import (
 
 	v1 "k8s.io/api/core/v1"
 	"k8s.io/apimachinery/pkg/api/resource"
+	corev1helpers "k8s.io/component-helpers/scheduling/corev1"
 )
 
 // PodRequestsAndLimits returns a dictionary of all defined resources summed up for all
-// containers of the pod. Pod overhead is added to the
-// total container resource requests and to the total container limits which have a
-// non-zero quantity.
+// containers of the pod. Pod overhead is added to the total container resource requests
+// and to the total container limits which have a non-zero quantity.  This may not be accurate for
+// scheduled pods as it does not take the InPlacePodVerticalScaling feature gate into consideration.
 func PodRequestsAndLimits(pod *v1.Pod) (reqs, limits v1.ResourceList) {
 	return PodRequestsAndLimitsReuse(pod, nil, nil)
 }
 
 // PodRequestsAndLimitsWithoutOverhead will create a dictionary of all defined resources summed up for all
-// containers of the pod.
+// containers of the pod. This may not be accurate for scheduled pods as it does not take the
+// InPlacePodVerticalScaling feature gate into consideration.
 func PodRequestsAndLimitsWithoutOverhead(pod *v1.Pod) (reqs, limits v1.ResourceList) {
-	reqs = make(v1.ResourceList, 4)
-	limits = make(v1.ResourceList, 4)
-	podRequestsAndLimitsWithoutOverhead(pod, reqs, limits)
-
-	return reqs, limits
-}
-
-func podRequestsAndLimitsWithoutOverhead(pod *v1.Pod, reqs, limits v1.ResourceList) {
-	for _, container := range pod.Spec.Containers {
-		addResourceList(reqs, container.Resources.Requests)
-		addResourceList(limits, container.Resources.Limits)
-	}
-	// init containers define the minimum of any resource
-	for _, container := range pod.Spec.InitContainers {
-		maxResourceList(reqs, container.Resources.Requests)
-		maxResourceList(limits, container.Resources.Limits)
+	opts := corev1helpers.PodResourcesOptions{
+		ExcludeOverhead: true,
 	}
+	return corev1helpers.PodRequests(pod, opts), corev1helpers.PodLimits(pod, opts)
```

Comment:
moving API validation out of API packages makes it too easy to accidentally change behavior in ways that expand or break our API surface. That's a key reason several of the functions lived in this package, unexported, so they wouldn't gain additional callers

Reply:
are any of the relocated functions used by API validation?

Reply:
if there are no API uses of the PodRequestsAndLimits / PodRequestsAndLimitsReuse / PodRequestsAndLimitsWithoutOverhead functions (for validation, etc), imo it would be clearer to move those out of this package

Reply:
Since these functions are not actually used in API validation, we don't need to block this PR, but it would be helpful to clarify they *shouldn't* be used in API validation to move them out of this package in a quick follow up

Reply:
Thanks for the prompt response. +1.

Reply:
We plan to use it here: https://github.com/kubernetes/kubernetes/pull/115643 in the new field

Reply:
> We plan to use it here: https://github.com/kubernetes/kubernetes/pull/115643 in the new field

if the intent is to use this in API validation, then the implementation should live in an API package, not in component-helpers, which has looser review requirements

/hold

sorry for the back and forth

Reply:
The reason to merging implementation is that it will become much more complex with sidecar containers KEP. So we want to keep it unified.

Reply:
@liggitt what would be preferred? Just move implementation here?

Reply:
ping me when the functions are relocated and I can ack

Reply:
@liggitt Relocation is complete, PTAL when you can.